app/src/main/java/com/ksale/otpview/ui/otpview/OtpView.kt

```diff
@@ -0,0 +1,274 @@
+) {
+    Box(modifier = modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+        TextField(
+            value = text,
+            onValueChange = {
+                if (it.length <= numberOfFields && it.isDigitsOnly()) {
+                    onValueChange(it)
+                }
+            },
+            modifier = Modifier
+                .size(0.dp)
+                .focusRequester(focusRequester)
+                .onFocusChanged {
+                    onFocusChanged(it.isFocused)
+                },
+            keyboardOptions = KeyboardOptions.Default.copy(
+                keyboardType = KeyboardType.Number
+            ),
+            keyboardActions = KeyboardActions(
+
+            )
+        )
+        Row(
+            modifier = Modifier
+                .fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            val nextFocusIndex = text.length
+            for (i in 0 until numberOfFields) {
+                PinField(
+                    modifier = Modifier,
+                    width = singlePinWidth,
+                    digit = text.getOrNull(i)?.toString() ?: "",
+                    textStyle = textStyle,
+                    shape = shape,
+                    isFocus = isFocus,
+                    isFilled = text.getOrNull(i) != null,
+                    isCurrentPinFocused = nextFocusIndex == i,
+                    isCursorEnabled = isCursorEnabled,
+                    isError = isError,
+                    defaultColor = defaultColor,
+                    focusColor = focusColor,
+                    filledColor = filledColor,
+                    cursorColor = cursorColor,
+                    errorColor = errorColor,
+                    drawStyle = drawStyle,
+                    index = i,
+                    onClick = onClick,
+                )
+            }
+        }
+    }
+}
+
+/**
+ * It is composable for displaying an individual pin of the OtpView
+ *
+ * @param modifier: Modifier for the Pin
+ * @param width: Width of the pin
+ * @param index: index of the pin
+ * @param digit: Digit in the particular pin
+ * @param textStyle: Style to be applied on the digit
+ */
+@Composable
+fun PinField(
+    modifier: Modifier = Modifier,
+    width: Dp,
+    index: Int,
+    digit: String,
+    textStyle: androidx.compose.ui.text.TextStyle,
+    isFocus: Boolean,
+    isFilled: Boolean,
+    isCurrentPinFocused: Boolean,
+    isCursorEnabled: Boolean,
+    isError: Boolean,
+    defaultColor: Color,
+    focusColor: Color,
+    filledColor: Color,
+    cursorColor: Color,
+    errorColor: Color,
+    shape: PinShape,
+    drawStyle: DrawStyle,
+    onClick: (index: Int) -> Unit,
+) {
+    val scope = rememberCoroutineScope()
+    val (cursorSymbol, setCursorSymbol) = remember { mutableStateOf("") }
+
+    if (isFocus && isCurrentPinFocused && !isFilled) {
+        LaunchedEffect(key1 = cursorSymbol, true) {
+            scope.launch {
```

Comment:
You don't need to call this in a separate scope. `LaunchedEffect`'s lambda gives us a CoroutineScope in the receiver context.

Reply:
The suspend function can be directly written inside LaunchedEffect